src/check_ast.rs

```diff
@@ -1191,10 +1193,13 @@ where
         // Pre-visit.
         match &expr.node {
             ExprKind::Subscript { value, slice, .. } => {
-                // Ex) typing.List[...]
+                // Ex) Optional[...]
                 if !self.in_deferred_string_annotation
                     && self.settings.enabled.contains(&CheckCode::U007)
-                    && self.settings.target_version >= PythonVersion::Py310
+                    && (self.settings.target_version >= PythonVersion::Py310
+                        || (self.settings.target_version >= PythonVersion::Py37
```

Comment:
`pyupgrade` seems to allow this on any Python version (if an annotation, with `__future__` enabled). I can't find a reference, so just keeping to 3.7 and above for now.